src/main/kotlin/org/opensearch/replication/action/stop/TransportStopIndexReplicationAction.kt

```diff
@@ -93,7 +98,13 @@ class TransportStopIndexReplicationAction @Inject constructor(transportService:
                     throw OpenSearchException("Failed to remove index block on ${request.indexName}")
                 }
 
-                validateReplicationStateOfIndex(request)
+                try {
+                    validateReplicationStateOfIndex(request)
+                }catch (e :Exception) {
+                    //Clear any stale replication tasks that maybe present in cluster state
+                    removeStaleReplicationTasksFromClusterState(request)
```

Comment:
We don't need this here, 
at the last stage of stop, we can wait for the tasks and trigger removal if required.

Reply:
It appears, we are already making one more call under finally, will this not address this? 


Looking at the code, we do want to execute stop based on all the checks related to metadata or any stale entries being present, can we not check for this and proceed with the execution? (rather than having to fail the validation check itself)